test/compress/issue-203.js

```diff
@@ -49,8 +49,8 @@ compress_new_function_with_destruct_arrows: {
         new Function("[[aa]], [{bb}]", 'return aa;');
     }
     expect: {
-        Function("n,[a]", "return n");
-        Function("b,{bb:n}", "return b");
-        Function("[[b]],[{bb:n}]", "return b");
+        Function("n,[r]", "return n");
+        Function("n,{bb:b}", "return n");
+        Function("[[n]],[{bb:b}]", "return n");
```

Comment:
Okay you need both `arrows` and `unsafe_arrows` - let me demonstrate what is happening with `unsafe_Func` below:
```diff
--- a/lib/compress.js
+++ b/lib/compress.js
@@ -84,6 +84,7 @@ function Compressor(options, false_by_default) {
         toplevel      : !!(options && options["top_retain"]),
         typeofs       : !false_by_default,
         unsafe        : false,
+        unsafe_arrows : false,
         unsafe_comps  : false,
         unsafe_Func   : false,
         unsafe_math   : false,
@@ -3505,6 +3506,7 @@ merge(Compressor.prototype, {
                     base54.reset();
                     ast.compute_char_frequency(mangle);
                     ast.mangle_names(mangle);
+console.error(ast.print_to_string());
                     var fun;
                     ast.walk(new TreeWalker(function(node) {
                         if (fun) return true;
@@ -4748,7 +4750,7 @@ merge(Compressor.prototype, {

     OPT(AST_Function, function(self, compressor){
         tighten_body(self.body, compressor);
-        if (compressor.option("arrows")
+        if (compressor.option("unsafe_arrows")
             && compressor.option("ecma") >= 6
             && !self.name
             && !self.is_generator
```
The above is against `harmony`, and it's basically this PR plus one `console.error()`
```sh
$ echo 'new Function("aa, [bb]", "return aa;");' | uglifyjs --ecma 6 -c unsafe,unsafe_Func,unsafe_arrows,arrows=0
n((n,[r])=>{return n});
Function("n,[r]","return n");

$ echo 'new Function("aa, [bb]", "return aa;");' | uglifyjs --ecma 6 -c unsafe,unsafe_Func,unsafe_arrows,arrows=1
n((n,[a])=>n);
Function("n,[a]","return n");
```
Essentially the presence of `return` due to lack of `arrows` was messing up character frequencies for `mangle`.